pkg/proxy/proxier.go

```diff
@@ -548,12 +555,33 @@ func (proxier *Proxier) openPortal(service types.NamespacedName, info *serviceIn
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIP {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
```

Comment:
I agree that it isn't great.  It looked like the Rackspace LB patch was doing the groundwork to make this better, and I didn't want to duplicate that effort.

ELB is tricksy: it has a dynamic set of load balancers.  It assigns them a public DNS name (something like myloadbalancer.elb.amazonaws.com), and you (as an ELB user) must CNAME your domain to it (or use Route 53 which integrates with ELB).

So there is no easy source IP address, sadly.  We do control the TCP port and could firewall that off, but I don't think that helps with the root issues here.

We have to publish the "myloadbalancer.elb.amazonaws.com" name so that the user can set up their CNAME.  It made sense to publish this in the publicIPs list.  And then that ended up in the info object.

We could explicitly recognize AWS, either with a flag to kube-proxy or by recognizing the format of the ELB name.  Would that be better?  Any other ideas?